src/main/java/com/github/princesslana/eriscasper/data/util/AuditLogChangeDeserializer.java

```diff
@@ -0,0 +1,165 @@
+package com.github.princesslana.eriscasper.data.util;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.github.princesslana.eriscasper.data.Data;
+import com.github.princesslana.eriscasper.data.DataException;
+import com.github.princesslana.eriscasper.data.Snowflake;
+import com.github.princesslana.eriscasper.data.resource.AuditLogChange;
+import com.github.princesslana.eriscasper.data.resource.ImmutableAuditLogChange;
+import com.github.princesslana.eriscasper.data.resource.Overwrite;
+import com.github.princesslana.eriscasper.data.resource.Role;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.Optional;
+
+public class AuditLogChangeDeserializer extends StdDeserializer<AuditLogChange> {
+
+  AuditLogChangeDeserializer() {
+    this(null);
+  }
+
+  private AuditLogChangeDeserializer(Class<?> vc) {
+    super(vc);
+  }
+
+  @Override
+  public AuditLogChange deserialize(JsonParser json, DeserializationContext context)
+      throws IOException {
+    JsonNode node = json.getCodec().readTree(json);
+    try {
+      return parseJson(node);
+    } catch (DataException e) {
+      e.printStackTrace();
```

Comment:
Let's not log to stdout.

Why are we returning a dummy object, instead of letting the exception bubble up?

Reply:
DataException needs to be handled and it won't let me pass it as a method throws due to me overriding the interface method. What could we do with the exception? Pass it through another exception?

Reply:
isn't there a standard json parsing exception or similar that deserializers should throw?